src/components/Card.js

```diff
@@ -15,39 +25,35 @@ export default class Card {
     return cardElement;
   }
 
-  _handleDeleteCard() {
-    this._card.remove();
-    this._card = null;
-  }
-
-  _handleLike() {
-    this._card
-      .querySelector(".card__like")
-      .classList.toggle("card__like_active");
-  }
-
   _setEventListeners() {
-    this._card.querySelector(".card__delete").addEventListener("click", () => {
-      this._handleDeleteCard();
+    this.card.querySelector(".card__delete").addEventListener("click", () => {
+      this._handleDeleteClick();
     });
-    this._card.querySelector(".card__like").addEventListener("click", () => {
-      this._handleLike();
+    this.card.querySelector(".card__like").addEventListener("click", () => {
```

Comment:
`this.card.querySelector(".card__like")`

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  generateCard() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```